appsec/src/helper/subscriber/waf.cpp

```diff
@@ -416,15 +420,58 @@ instance::listener::~listener()
 void instance::listener::call(
     dds::parameter_view &data, event &event, const std::string &rasp_rule)
 {
-    ddwaf_result res;
+    ddwaf_object res;
     DDWAF_RET_CODE code;
+    unsigned duration = 0;
+    bool timeout = false;
+    const ddwaf_object *events = nullptr;
+    const ddwaf_object *actions = nullptr;
+    const ddwaf_object *attributes = nullptr;
     auto run_waf = [&]() {
         dds::parameter_view *persistent = rasp_rule.empty() ? &data : nullptr;
         dds::parameter_view *ephemeral = rasp_rule.empty() ? nullptr : &data;
         code = ddwaf_run(
             handle_, persistent, ephemeral, &res, waf_timeout_.count());
-    };
+        for (size_t i = 0; i < ddwaf_object_size(&res); ++i) {
+            const ddwaf_object *child = ddwaf_object_get_index(&res, i);
+            if (child == nullptr) {
+                continue;
+            }
 
+            size_t length = 0;
+            const char *key = ddwaf_object_get_key(child, &length);
+            if (key == nullptr) {
+                continue;
+            }
+
+            static constexpr std::string_view events_str = "events";
+            static constexpr std::string_view actions_str = "actions";
+            static constexpr std::string_view attributes_str = "attributes";
+            static constexpr std::string_view keep_str = "keep";
+            static constexpr std::string_view duration_str = "duration";
+            static constexpr std::string_view timeout_str = "timeout";
+
```

Comment:
You can assign the key to a string view and use a more natural comparison, for example:

```cpp
std::string_view key_sv{key, length};
if (key_sv == "events"sv) { ... }
```